### PR TITLE
Feature/internal view edit by cy

### PR DESCRIPTION
**Pull request by SumeN**
For PBI - 73689. I am yet to do task 73818 (Only allow delete on the latest record)

### CODE REVIEW

File: src/EA.Weee.RequestHandlers.Tests.Unit/AatfReturn/Internal/AatfDataAccessTests.cs [hunk 1]
```diff
@@ -354,5 +354,37 @@ public async void DeleteAatf_DeletesAatf()
                 .Then(A.CallTo(() => genericDataAccess.Remove(returnAatf)).MustHaveHappened(Repeated.Exactly.Once))
                 .Then(A.CallTo(() => context.SaveChangesAsync()).MustHaveHappened(Repeated.Exactly.Once));
         }
+
+        [Fact]
+        public async Task GetDetails_GivenAatfId_ComplianceYearsShouldBeReturned()
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.RequestHandlers.Tests.Unit/AatfReturn/Internal/AatfDataAccessTests.cs]: Do test where aatf id does not match no result should be returns

File: src/EA.Weee.RequestHandlers/AatfReturn/Internal/AatfDataAccess.cs [hunk 2]
```diff
@@ -216,5 +216,27 @@ public async Task RemoveAatfData(Aatf aatf, IEnumerable<int> quarters)
...
+        {
+            return await context.Aatfs
+                    .Where(r => r.AatfId == aatfId)
+                    .Select(r => r.ComplianceYear)
+                    .Distinct()
+                    .OrderByDescending(year => year)
+                    .ToListAsync();
+        }
+
+        public async Task<Guid> GetAatfId(Guid aatfId, short complianceYear)
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.RequestHandlers/AatfReturn/Internal/AatfDataAccess.cs]: Rename this to GetAatfByAatfIdAndComplianceYear

File: src/EA.Weee.Web/Areas/Admin/Controllers/AatfController.cs [hunk 3]
```diff
@@ -74,7 +76,8 @@ public async Task<ActionResult> Details(Guid id)
                     OrganisationString = GenerateSharedAddress(aatf.Organisation.BusinessAddress),
                     AssociatedAatfs = associatedAatfs,
                     AssociatedSchemes = associatedSchemes,
-                    SubmissionHistory = submissionHistory
+                    SubmissionHistory = submissionHistory,
+                    ComplianceYearList = years
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Admin/Controllers/AatfController.cs]: Update the DetailsGet_GivenValidAatfId_MapperShouldBeCalled test to check the mapper s called with this list

File: src/EA.Weee.Web/Areas/Admin/Mappings/ToViewModel/AatfDataToAatfDetailsViewModelMap.cs [hunk 4]
```diff
@@ -44,7 +44,10 @@ public AatfDetailsViewModel Map(AatfDataToAatfDetailsViewModelMapTransfer source
                 SiteAddressLong = addressUtilities.FormattedAddress(source.AatfData.SiteAddress, false),
                 ContactAddressLong = addressUtilities.FormattedAddress(source.AatfData.Contact.AddressData, false),
                 PanArea = source.AatfData.PanAreaData,
-                LocalArea = source.AatfData.LocalAreaData
+                LocalArea = source.AatfData.LocalAreaData,
+                ComplianceYearList = source.ComplianceYearList,
+                AatfId = source.AatfData.AatfId,
+                SelectedComplianceYear = source.AatfData.ComplianceYear
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Admin/Mappings/ToViewModel/AatfDataToAatfDetailsViewModelMap.cs]: Update test to check selected compliance year is set

**Comment by JohnPadgettCivica** [comment]
The data tab should also be filtered by year, so need to update GetAatfSubmissionHistory request to take year and filter in the stored procedure

AC point 4 there is some static text that needs adding

File: src/EA.Weee.RequestHandlers.Tests.Unit/AatfReturn/Internal/AatfDataAccessTests.cs (hunk 1 above)
Comment (SumeN) [reply src/EA.Weee.RequestHandlers.Tests.Unit/AatfReturn/Internal/AatfDataAccessTests.cs]: Added

File: src/EA.Weee.RequestHandlers/AatfReturn/Internal/AatfDataAccess.cs (hunk 2 above)
Comment (SumeN) [reply src/EA.Weee.RequestHandlers/AatfReturn/Internal/AatfDataAccess.cs]: Renamed

File: src/EA.Weee.Web/Areas/Admin/Controllers/AatfController.cs (hunk 3 above)
Comment (SumeN) [reply src/EA.Weee.Web/Areas/Admin/Controllers/AatfController.cs]: Added

File: src/EA.Weee.Web/Areas/Admin/Mappings/ToViewModel/AatfDataToAatfDetailsViewModelMap.cs (hunk 4 above)
Comment (SumeN) [reply src/EA.Weee.Web/Areas/Admin/Mappings/ToViewModel/AatfDataToAatfDetailsViewModelMap.cs]: Done

**Comment by SumeN** [comment]
Passed to year to SP in the data access method

**Comment by JohnPadgettCivica** [comment]
Update GetAatfSubmissionsTests and GetAeSubmissionsTests to have tests to check that data from the correct year is coming back

Also remove the year field from the data tab, also static text to be added to the edit screen

**Pull request merged**